src/Enemy/EnemyStateHackStart.cpp

```diff
@@ -0,0 +1,176 @@
+#include "Enemy/EnemyStateHackStart.h"
+#include <al/Library/LiveActor/ActorActionFunction.h>
+#include <al/Library/LiveActor/ActorAnimFunction.h>
+#include <al/Library/LiveActor/ActorFlagFunction.h>
+#include <al/Library/LiveActor/ActorMovementFunction.h>
+#include <al/Library/LiveActor/LiveActorFunction.h>
+#include <al/Library/Nerve/NerveSetupUtil.h>
+#include <al/Library/Nerve/NerveUtil.h>
+#include <al/Library/Shadow/DepthShadowDrawer.h>
+#include "Player/PlayerHackKeeper.h"
+#include "Player/PlayerHackStartShaderCtrl.h"
+#include "Util/Hack.h"
+
+namespace rs {
+bool isMsgStartHack(const al::SensorMsg*);
+}
+
+namespace {
+NERVE_IMPL(EnemyStateHackStart, DiveIn);
+NERVE_IMPL(EnemyStateHackStart, HackStart);
+
+NERVE_MAKE(EnemyStateHackStart, DiveIn);
+NERVE_MAKE(EnemyStateHackStart, HackStart);
+}  // namespace
+
+EnemyStateHackStartParam::EnemyStateHackStartParam(const char* actionName, const char* visAnimName, const char* mtpAnimName, bool hasSubActors,
+                                                   bool updateSubActorShadowMap) {
+    mActionName = actionName;
+    mVisAnimName = visAnimName;
+    mMtpAnimName = mtpAnimName;
+    mHasSubActors = hasSubActors;
+    mUpdateSubActorShadowMap = updateSubActorShadowMap;
+}
+
+static EnemyStateHackStartParam fallbackParam("HackStart", 0, 0, 0, 0);
+
+EnemyStateHackStart::EnemyStateHackStart(al::LiveActor* rootActor, const EnemyStateHackStartParam* param, PlayerHackStartShaderParam* shaderParam)
+    : al::ActorStateBase("憑依開始", rootActor) {
+    if (param) {
+        mParam = param;
+    } else {
+        mParam = &fallbackParam;
+    }
```

Comment:
```suggestion
    mParam = param ? param : &fallbackParam;
```
That's open for discussion though, might be too short/unreadable.

Reply:
I think it's a little unreadable that way. I have now flipped it though so it checks for !param first, to set as fallbackParam instead.

Reply:
I'm not sure how I missed it the first time, but this is **the** use case for the elvis operator.
```cpp
mParam = param ?: &fallbackParam;
```

Reply:
This mismatches, your implementations put `ne` instead of `eq` on the csel

Reply:
Can you create a `decomp.me` of it?